### ISSUE TITLE
Spring causes rspec to always order tests the same

### ISSUE THREAD

**Issue by dipth** [rails/spring#113]
When running rspec with `config.order = "random"` spring causes the order to always be the same.

```
[master●●] % spring rspec
....

Finished in 0.21176 seconds
4 examples, 0 failures

Randomized with seed 6661

[master●●] % spring rspec
....

Finished in 0.21373 seconds
4 examples, 0 failures

Randomized with seed 6661

[master●●] % spring rspec
....

Finished in 0.16834 seconds
4 examples, 0 failures

Randomized with seed 6661
```

**Comment by Aqualon** [comment]
The workaround with setting config.seed to random value in spec_helper works, but it prevents running specs with the --seed parameter. So not a real solution.

**Comment by draffensperger** [comment]
Yeah, I had the same issue of wanting to be able to set the seed, but also have it not get stuck when I'm using spring. It turns out that rspec supports another way of setting the seed, ` --order=rand:[seed]`, so you can run a spec like this `bin/rspec --order=rand:2 spec/models/my_model.rb` and it will use a seed of `2` even if you have `config.seed = srand % 0xFFFF` in your spec helper.

Ideally I'd like the `--seed` parameter to work since that seems more intuitive, but that's a workaround at least.

**Comment by mwean** [comment]
I'm also having this problem. Could we re-open this issue?

**Comment by mwean** [comment]
For what it's worth, this is what I'm doing now in my `spec_helper.rb`:

`config.seed = srand % 0xFFFF unless ARGV.any? { |arg| arg =~ /seed/ }`

This still gives me the ability to override the seed like I'm used to.

**Comment by deivid-rodriguez** [comment]
Thanks @mwean, your workaround works perfectly! @jonleighton This issue should probably be reopened.

**Comment by soberstadt** [comment]
I am also having this issue.

**Comment by CUnknown** [comment]
This still seems to be an issue [spring@1.6.3, rspec@3.4.0] but the suggestion of @mwean works :D

**Comment by TylerRick** [comment]
Just ran into this on spring@2.0.2, rspec@3.7.0. Thankfully the workaround above still works. 

Seems like it should just work out of the box. Where would the fix go though? In `spring`? In `spring-commands-rspec`?

**Comment by javierjulio** [comment]
I've noticed the same issue today. Since its tied to `spring` I didn't want to include it directly in `spec_helper` so I ended up combining the solutions from @jcarlson and @mwean into one in the `config/spring.rb` file:

```ruby
Spring.after_fork do
  if Rails.env == 'test'
    RSpec.configure do |config|
      config.seed = srand % 0xFFFF unless ARGV.any? { |arg| arg =~ /seed/ }
    end
  end
end
```

This way it's only done if someone is running `bin/rspec` but still supports overriding the seed through the command line parameter. Thanks!
